segmentstore/server/src/main/java/io/pravega/segmentstore/server/security/TLSHelper.java

```diff
@@ -0,0 +1,96 @@
+
+    /**
+     * Creates a new instance of {@link SslContext}.
+     *
+     * @param pathToCertificateFile the path to the PEM-encoded server certificate file
+     * @param pathToServerKeyFile the path to the PEM-encoded file containing the server's encrypted private key
+     * @return a {@link SslContext} built from the specified {@code pathToCertificateFile} and {@code pathToServerKeyFile}
+     * @throws NullPointerException if either {@code pathToCertificateFile} or {@code pathToServerKeyFile} is null
+     * @throws IllegalArgumentException if either {@code pathToCertificateFile} or {@code pathToServerKeyFile} is empty
+     * @throws RuntimeException if there is a failure in building the {@link SslContext}
+     */
+    public static SslContext newServerSslContext(String pathToCertificateFile, String pathToServerKeyFile) {
+        Exceptions.checkNotNullOrEmpty(pathToCertificateFile, "pathToCertificateFile");
+        Exceptions.checkNotNullOrEmpty(pathToServerKeyFile, "pathToServerKeyFile");
+        return newServerSslContext(new File(pathToCertificateFile), new File(pathToServerKeyFile));
+    }
+
+    /**
+     * Creates a new instance of {@link SslContext}.
+     *
+     * @param certificateFile the PEM-encoded server certificate file
+     * @param serverKeyFile the PEM-encoded file containing the server's encrypted private key
+     * @return a {@link SslContext} built from the specified {@code pathToCertificateFile} and {@code pathToServerKeyFile}
+     * @throws NullPointerException if either {@code certificateFile} or {@code serverKeyFile} is null
+     * @throws IllegalStateException if either {@code certificateFile} or {@code serverKeyFile} doesn't exist or is unreadable.
+     * @throws RuntimeException if there is a failure in building the {@link SslContext}
+     */
+    public static SslContext newServerSslContext(File certificateFile, File serverKeyFile) {
+        Preconditions.checkNotNull(certificateFile);
+        Preconditions.checkNotNull(serverKeyFile);
+
+        Retry.RetryAndThrowExceptionally<IllegalStateException, Exception> retryConfig =
+                Retry.withExpBackoff(1, 10, 3, 20000)
+                        .retryingOn(IllegalStateException.class)
+                        .throwingOn(Exception.class);
+
+        ensureExistAndAreReadable(certificateFile, serverKeyFile);
+
+        try {
+            SslContext result = null;
+            result = SslContextBuilder.forServer(certificateFile, serverKeyFile).build();
+            log.debug("Done creating a new SSL Context for the server.");
+            return result;
+        } catch (SSLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void ensureExistAndAreReadable(File certificateFile, File serverKeyFile) {
+        if (!certificateFile.exists()) {
+            throw new IllegalStateException(String.format("Certificate file %s doesn't exist",
```

Comment:
These look a lot more like `IllegalArgumentExceptions` since you are validating your arguments, not the internal state of your object.

I would also use `Preconditions.checkArgument` to make your code more compact.

Reply:
Yes, that makes sense. I've modified the code accordingly.